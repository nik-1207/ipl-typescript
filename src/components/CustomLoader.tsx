import { useEffect, useState } from "react";
import Loader from "react-loader-spinner";
import { LoadingEvent } from "../event/LoadingEvent";
import { subscribe } from "../config/PubSub";

export default function CustomLoader() {
  const [loading, setloading] = useState();
  useEffect(() => {
    const handle = subscribe(LoadingEvent, (args: any) => {
      setloading(args.isLoading);
    });
    return () => {
      handle.unsubscribe();
    };
  }, []);

  return (
    <>
      {loading && (
        <div className={"loader"}>
            <Loader
          type="TailSpin"
          color="#00BFFF"
          height={80}
          width={80}
        />
        </div>
      )}
    </>
  );
}
