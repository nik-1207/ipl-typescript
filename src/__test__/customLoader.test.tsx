import renderer from "react-test-renderer";
import CustomLoader from "../components/CustomLoader";
it("customLoader snapshot testing", () => {
  const comp = renderer.create(<CustomLoader />);
  const snap = comp.toJSON();
  expect(snap).toMatchSnapshot();
});
