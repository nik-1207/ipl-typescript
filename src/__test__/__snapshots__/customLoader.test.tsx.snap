// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customLoader snapshot testing 1`] = `null`;
